katana/data/wrf_out.py

```diff
@@ -0,0 +1,189 @@
+            topo {Topo class} -- katana.data.topo.Topo class instance
+        """
+
+        self._logger = logging.getLogger(__name__)
+
+        self.config = config
+        self.topo = topo
+
+        self.start_date = self.config['time']['start_date']
+        self.end_date = self.config['time']['end_date']
+
+        # create an hourly time step between the start date and end date
+        self.date_list = utils.daterange(self.start_date, self.end_date)
+
+        # create a daily list between the start and end date
+        self.day_list = utils.daylist(self.start_date, self.end_date)
+
+        # WRF file information
+        self.wrf_filename = self.config['input']['wrf_filename']
+        self.get_wrf_time_info()
+
+        self.out_dir = self.config['output']['out_location']
+        self.out_dir_tmp = os.path.join(self.out_dir, 'tmp')
+
+        self._logger.debug('WRFout initialized')
+
+    def get_wrf_time_info(self):
+        """Retrieve information from the WRF file as to the time
+        range of the file
+        """
+
+        f = nc.Dataset(self.wrf_filename)
+        t = f.variables['Times']
+
+        # borrowed from SMRF
+        t.set_auto_maskandscale(False)
+        try:
+            times = [('').join(v) for v in t]
+        except TypeError:
+            times = []
+            for v in t:
+                times.append(''.join([s.decode('utf-8') for s in v]))
+        except Exception:
+            raise Exception('Could not convert WRF times to readable format')
+
+        # remove the underscore and convert to datetime object
+        times = [utils.parse_date(v.replace('_', ' '))
+                 for v in times]
+
+        # set the timezone
+        self.wrf_times = [pytz.utc.localize(t) for t in times]
+
+        f.close()
+
+        self.check_file_dates()
+
+    def check_file_dates(self):
+        """Check that the WRF file dates coorespond to the dates provided
+        """
+
+        # check that the dates are within the file range
+        in_between_dates = []
+        for d in self.wrf_times:
+            if d >= self.start_date and d <= self.end_date:
+                in_between_dates.append(d)
+
+        num_dates = len(in_between_dates)
+        num_list = len(self.date_list)
+        num_wrf = len(self.wrf_times)
+
+        if num_dates != num_list:
+            self._logger.warning('Not all dates present in WRF file')
+
+        if num_dates == 0:
+            self._logger.error(
+                'No times in WRF file between the start and end date')
+
+        if num_wrf > num_list:
+            self._logger.warning("There are {} more times in the WRF file "
+                                 "than specified. WindNinja will perform "
+                                 "simulations for all times in file".format(
+                                     num_wrf - num_list))
+
+    def initialize_data(self):
+        """Initialize the data if needed
+        """
+        pass
```

Comment:
Could be moved to `BaseData`. Then it is ensured that all inheriting classes have It and they can overwrite it there.